cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/NewTwoWindingsTransformerConversion.java

```diff
@@ -0,0 +1,426 @@
+        ConvertedT2xModel convertedT2xModel = convertToIidm(interpretedT2xModel);
+
+        setToIidm(convertedT2xModel);
+    }
+
+    private CgmesT2xModel load() {
+        // ends = ps
+        PropertyBag end1 = ps.get(0);
+        PropertyBag end2 = ps.get(1);
+
+        double x1 = end1.asDouble(STRING_X);
+        double x2 = end2.asDouble(STRING_X);
+        double r = end1.asDouble(STRING_R) + end2.asDouble(STRING_R);
+        double x = x1 + x2;
+
+        String terminal1 = end1.getId(CgmesNames.TERMINAL);
+        String terminal2 = end2.getId(CgmesNames.TERMINAL);
+
+        PropertyBag rtc1 = getTransformerTapChanger(end1, STRING_RATIO_TAP_CHANGER, powerTransformerRatioTapChanger);
+        PropertyBag ptc1 = getTransformerTapChanger(end1, STRING_PHASE_TAP_CHANGER, powerTransformerPhaseTapChanger);
+        PropertyBag rtc2 = getTransformerTapChanger(end2, STRING_RATIO_TAP_CHANGER, powerTransformerRatioTapChanger);
+        PropertyBag ptc2 = getTransformerTapChanger(end2, STRING_PHASE_TAP_CHANGER, powerTransformerPhaseTapChanger);
+
+        double ratedU1 = end1.asDouble(STRING_RATEDU);
+        double ratedU2 = end2.asDouble(STRING_RATEDU);
+
+        TapChangerConversion ratioTapChanger1 = getRatioTapChanger(rtc1);
+        TapChangerConversion ratioTapChanger2 = getRatioTapChanger(rtc2);
+        TapChangerConversion phaseTapChanger1 = getPhaseTapChanger(ptc1, x);
+        TapChangerConversion phaseTapChanger2 = getPhaseTapChanger(ptc2, x);
+
+        CgmesT2xModel cgmesT2xModel = new CgmesT2xModel();
+        cgmesT2xModel.end1.g = end1.asDouble(STRING_G, 0);
+        cgmesT2xModel.end1.b = end1.asDouble(STRING_B);
+        cgmesT2xModel.end1.ratioTapChanger = ratioTapChanger1;
+        cgmesT2xModel.end1.phaseTapChanger = phaseTapChanger1;
+        cgmesT2xModel.end1.ratedU = ratedU1;
+        cgmesT2xModel.end1.phaseAngleClock = end1.asInt(STRING_PHASE_ANGLE_CLOCK, 0);
+        cgmesT2xModel.end1.terminal = terminal1;
+
+        if (x1 == 0.0) {
+            cgmesT2xModel.end1.xIsZero = true;
+        } else {
+            cgmesT2xModel.end1.xIsZero = false;
+        }
+
+        cgmesT2xModel.end2.g = end2.asDouble(STRING_G, 0);
+        cgmesT2xModel.end2.b = end2.asDouble(STRING_B);
+        cgmesT2xModel.end2.ratioTapChanger = ratioTapChanger2;
+        cgmesT2xModel.end2.phaseTapChanger = phaseTapChanger2;
+        cgmesT2xModel.end2.ratedU = ratedU2;
+        cgmesT2xModel.end2.phaseAngleClock = end2.asInt(STRING_PHASE_ANGLE_CLOCK, 0);
+        cgmesT2xModel.end2.terminal = terminal2;
+
+        if (x2 == 0.0) {
+            cgmesT2xModel.end2.xIsZero = true;
+        } else {
+            cgmesT2xModel.end2.xIsZero = false;
+        }
+
+        cgmesT2xModel.r = r;
+        cgmesT2xModel.x = x;
+
+        return cgmesT2xModel;
+    }
+
+    private InterpretedT2xModel interpret(CgmesT2xModel cgmesT2xModel, Conversion.Config alternative) {
```

Comment:
For each method call below, explain what it is being done (checks and possible interpretations)

Reply:
An initial documentation has been added to the main methods and at class level.